### PR TITLE
fix NA for avgArmAngelAbsDiff with epoch size 5sec

### PR DESCRIPTION
**Pull request by angerhang**
`avgArmAngelAbsDiff` should be ill-defined when one epoch has less than 10 sec as it is supposed to be the absolute difference between the angel z for two 5-sec window. Returning 0 to make for practical purpose instead of NA.

Fixed the issue mentioned in https://github.com/activityMonitoring/biobankAccelerometerAnalysis/issues/103.

### CODE REVIEW

File: java/Features.java [hunk 1]
```diff
@@ -651,10 +652,12 @@ private static String getUnileverFeaturesHeader(){
...
-            double[] absoluteAvgDiff = computeAbsoluteDiff(fiveSecAvg);
-
-            // get the avg of difference
-            double avgArmAngelAbsDiff = AccStats.mean(absoluteAvgDiff);
+            //    ill-defined when we have fewer 10 sec for the epoch length
+            double avgArmAngelAbsDiff = 0;
+            if (x.length/sampleRate >= 10) {
+                double[] absoluteAvgDiff = computeAbsoluteDiff(fiveSecAvg);
+                avgArmAngelAbsDiff = AccStats.mean(absoluteAvgDiff);
+            }
```
Comment (chanshing) [inline java/Features.java:660]: Shall we also explain this at the top of the function? Currently it says

`This won't work if the epoch length is less than 5-sec. So return 0 in that case.`

If I understand correctly `avgArmAngel` requires >=5, but `avgArmAngelDiff` requires >=10 right?

**Review by chanshing — CHANGES_REQUESTED** [review]
Thanks Hang. Hope my minor comment helps.

File: java/Features.java (hunk 1 above)
Comment (angerhang) [reply java/Features.java:660]: Yes that's correct. Okay let me perhaps explain it a bit more.

**Comment by R-Walmsley** [comment]
Hi @angerhang , Looks fine to me - though I'm not at all familiar with this end of the code, or with Java...

**Comment by chanshing** [comment]
Thanks Hang. Finally, do you mind squashing both commits? After that should be ready to go :dancing_men:

**Comment by angerhang** [comment]
> Thanks Hang. Finally, do you mind squashing both commits? After that should be ready to go 👯‍♂️

Done ;D

**Comment by chanshing** [comment]
Thanks Hang. Best practice on commit messages from what I read is to use imperative tense ( "fix X" instead of "fixed X", "update X" instead of "updated X") https://www.danclarke.com/git-tense
Just a minor thing in case you feel like you need to force-push someone.
Otherwise, feel free to merge :dancing_men:

**Review by chanshing — APPROVED** [review]

**Pull request merged**